auth0/resource_auth0_client.go

```diff
@@ -330,3 +325,50 @@ func buildClient(d *schema.ResourceData) *management.Client {
 
 	return c
 }
+
+func buildSchema(c *management.Client, d *schema.ResourceData) {
```

Comment:
We should probably name this `buildClientSchema` so we can re-use the convention for other resources.

Reply:
After reading other providers, it seems the `readResource` should be responsible for this sort of thing. Essentially moving all the `d.Set(<key>, <value>)`'s inside the `[readClient](https://github.com/terraform-providers/terraform-provider-github/blob/master/github/resource_github_repository.go#L215-L233)` function and calling it after every `createClient` and `[updateClient](https://github.com/terraform-providers/terraform-provider-github/blob/master/github/resource_github_repository.go#L266)`.

I did the same in #13, so you can get a better idea of what I mean